app/models/manager_refresh/save_collection/helper.rb

```diff
@@ -34,8 +34,9 @@ def save_inventory_collection!(inventory_collection, association)
         inventory_objects_index[index] = inventory_object
       end
 
-      unique_index_keys = inventory_collection.manager_ref_to_cols
-      unique_db_indexes = Set.new
+      unique_index_keys      = inventory_collection.manager_ref_to_cols
+      unique_db_indexes      = Set.new
+      unique_db_primary_keys = Set.new
```

Comment:
on naming: I thought these are Primary Keys as in a [set of attributes of a relation variable](https://en.wikipedia.org/wiki/Unique_key). Like above `uniqe_index_keys`.

Maybe call this `visited_db_ids` ?